mvx-esdt-safe/src/register_token.rs

```diff
@@ -25,8 +25,7 @@ pub trait RegisterTokenModule:
         self.require_sov_token_id_not_registered(&sov_token_id);
 
         require!(
-            self.has_prefix(&sov_token_id)
-                || (self.is_native_token(&sov_token_id) && token_type == EsdtTokenType::Fungible),
+            self.has_prefix(&sov_token_id) && !self.is_native_token(&sov_token_id),
```

Comment:
I think this is enough. Native token for sure doesn't have prefix because it was issued on main chain.
```suggestion
            self.has_prefix(&sov_token_id)
```